Automattic/jetpack
# Connection: trigger "Restore Connection" flow

#### Changes proposed in this Pull Request:
* Trigger restore connection flow via the connection error notice

#### Other information:


#### Jetpack product discussion
p9dueE-5zL-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
_This PR can no longer be tested, because we removed the My Jetpack integration.
Please look at the code and see if it makes sense, and confirm Jetpack Connection still works
No actual testing needed._

_SQL query to remove transients:_
```delete from wp_options where option_name like '%jetpack_connection_error_reporting_gate_%';```

1. Connect Jetpack, use "Broken Token" to set invalid blog and user tokens.
2. Go to "Jetpack -> My Jetpack", you should see the error. If it doesn't show up, remove the transients using the SQL query above. <img width="1096" alt="Screen Shot 2022-09-28 at 17 18 03" src="https://user-images.githubusercontent.com/1341249/192890640-f9a34672-0fee-4925-a039-9691c7895ca8.png">
3. Click "Restore Connection", the notice should say "Reconnecting Jetpack":  <img width="292" alt="Screen Shot 2022-09-28 at 17 18 41" src="https://user-images.githubusercontent.com/1341249/192890721-ddebd717-8df4-4b21-a0a5-e562cb2c44af.png">
4. The connection screen should show up. Finish the connection flow, confirm it works well.
5. Go back to "Broken Token" and break only the blog token, leaving the user token intact.
6. Remove the transients using the SQL query above.
7. Go to "Jetpack -> My Jetpack", you should see the same error notice. Click "Restore".
8. The same "Restoring Jetpack" notice should show up.
9. After a few seconds, the page should reload. The error notice should no longer appear.
10. Go to "Broken Token" and confirm that the blog token has been restored.
